### DISCUSSION TITLE
How does everyone deal with tree shadows?

### DISCUSSION THREAD

**Discussion by superdong69** [blakeblackshear/frigate#6574] [Ask A Question] [answered]
So this is my first time with leaves on the trees since I've setup Frigate.
How do I stop Frigate ( with coral ) from thinking shadows on my driveway are Persons?

https://ibb.co/GMSzdKJ

**Comment by NickM-27** [comment]
This is one of the many reasons for https://docs.frigate.video/integrations/plus so there will be better models that don't get tripped up by shadows like this.

In any case, there are a number of filters that can be applied: https://docs.frigate.video/guides/false_positives

**Reply by superdong69** [reply]
I've seen the bottom link and read it, but it doesn't really explain how to use it properly. I have settings for persons for that screenshot:

`      filters:
        person:
          min_score: 0.4
          threshold: 0.6
          min_area: 900
`

I'm not sure what I need to change to get it to not see people in shadows.
When I have it set to 0.8 for threshold, I never get any positives when people are actually IN the frame, hence why I had to lower it. BUt even still, at 0.6, it thinks tree shadows are persons.

**Reply by NickM-27** [reply, marked as answer]
Yes, that is the issue. 0.7 is the default and you probably want to leave it there.

**Reply by superdong69** [reply]
Roger that, I'll change it back and see how I get along. Thanks

**Reply by blakeblackshear** [reply]
If that doesn't work, try bumping the min_score up to 0.6 without changing the threshold.

**Reply by nijave** [reply]
I know this is old but it still comes up as a top search result. The Frigate Plus models work much better ~and afaik they recently released the base model for free~ (edit: see comment below)

You can also try playing with motion/object masks which helped for regions with trees that were triggering detections and portions of the camera view field that may contain shadows but you don't actually care about (like near the edges of the frame that might be street).

**Reply by NickM-27** [reply]
> afaik they recently released the base model for free.

this is incorrect, the base model is now directly available to subscribers, no Frigate+ models are free

**Comment by skrashevich** [comment]
you can try to use mode accurate model:
download https://raw.githubusercontent.com/google-coral/test_data/master/efficientdet_lite3_512_ptq_edgetpu.tflite and https://raw.githubusercontent.com/google-coral/test_data/master/coco_labels.txt to the configuration directory, and put this lines to config;
```
model:
  path: /config/efficientdet_lite3_512_ptq_edgetpu.tflite
  labelmap_path: /config/coco_labels.txt
  width: 512
  height: 512
```

**Reply by superdong69** [reply]
Do I add that code to the: 

`detectors:
  coral:
    type: edgetpu
    device: usb
`

Section of the frigate.yaml?

**Reply by skrashevich** [reply]
Add this code to the top level of config

**Reply by superdong69** [reply]
Stupid question, where do those two files go? I know you said configuration directory, but not sure which that is from the docker compose on a stand alone linux distro.

**Reply by skrashevich** [reply]
These files should be in the same directory as the config.yml, frigate.db files...

**Reply by NickM-27** [reply]
You can put it wherever you want, just need to make sure the config and location match.

**Reply by superdong69** [reply]
Copied both files to /home/user/docker-compose/frigate/ and I get this error:

`frigate  | 2023-05-25 23:00:01.785556031  [Errno 2] No such file or directory: '/home/user/docker-compose/frigate/coco_labels.txt'
frigate  | 2023-05-25 23:00:01.786247860  Traceback (most recent call last):
frigate  | 2023-05-25 23:00:01.786249923    File "/opt/frigate/frigate/app.py", line 351, in start
frigate  | 2023-05-25 23:00:01.786251088      self.init_config()
frigate  | 2023-05-25 23:00:01.786252290    File "/opt/frigate/frigate/app.py", line 80, in init_config
frigate  | 2023-05-25 23:00:01.786253381      user_config = FrigateConfig.parse_file(config_file)
frigate  | 2023-05-25 23:00:01.786254475    File "/opt/frigate/frigate/config.py", line 1072, in parse_file
frigate  | 2023-05-25 23:00:01.786255455      return cls.parse_obj(config)
frigate  | 2023-05-25 23:00:01.786256492    File "pydantic/main.py", line 526, in pydantic.main.BaseModel.parse_obj
frigate  | 2023-05-25 23:00:01.786257553    File "pydantic/main.py", line 339, in pydantic.main.BaseModel.__init__
frigate  | 2023-05-25 23:00:01.786258610    File "pydantic/main.py", line 1076, in pydantic.main.validate_model
frigate  | 2023-05-25 23:00:01.786260458    File "pydantic/fields.py", line 884, in pydantic.fields.ModelField.validate
frigate  | 2023-05-25 23:00:01.786261629    File "pydantic/fields.py", line 1101, in pydantic.fields.ModelField._validate_singleton
frigate  | 2023-05-25 23:00:01.786292043    File "pydantic/fields.py", line 1151, in pydantic.fields.ModelField._apply_validators
frigate  | 2023-05-25 23:00:01.786293435    File "pydantic/class_validators.py", line 337, in pydantic.class_validators._generic_validator_basic.lambda13
frigate  | 2023-05-25 23:00:01.786299339    File "pydantic/main.py", line 711, in pydantic.main.BaseModel.validate
frigate  | 2023-05-25 23:00:01.786300514    File "/opt/frigate/frigate/detectors/detector_config.py", line 65, in __init__
frigate  | 2023-05-25 23:00:01.786301647      **load_labels(config.get("labelmap_path", "/labelmap.txt")),
frigate  | 2023-05-25 23:00:01.786302682    File "/opt/frigate/frigate/util.py", line 709, in load_labels
frigate  | 2023-05-25 23:00:01.786303689      with open(path, "r", encoding=encoding) as f:
frigate  | 2023-05-25 23:00:01.786309885  FileNotFoundError: [Errno 2] No such file or directory: '/home/user/docker-compose/frigate/coco_labels.txt`

And I have no idea why posting code in the code tags doesn't format the code properly. This site is so infuriating.

**Reply by skrashevich** [reply]
show output of `ls -la /home/user/docker-compose/frigate/` and the `model` section of frigate config 

keep in mind, all paths in config must be relative to frigate container, not to host machine

**Reply by superdong69** [reply]
`drwxrwxr-x 2 user user     4096 May 25 22:55 .`
`drwxrwxr-x 3 user user     4096 May 25 22:55 ..`
`-rw-rw-r-- 1 user user      661 May 23 18:08 coco_labels.txt`
`-rw-rw-r-- 1 user user      852 May 22 13:28 docker-compose.yml`
`-rw-rw-r-- 1 user user 15134032 May 23 18:08 efficientdet_lite3_512_ptq_edgetpu.tflite`
`-rw-rw-r-- 1 user user     6187 May 25 23:04 frigate.yml`

**Reply by superdong69** [reply]
`ui:`
 ` use_experimental: false`
  `live_mode: mse`
`mqtt:`
 ` host: 192.168.1.80`
  `port: 1883`
  `topic_prefix: frigate`
  `client_id: frigate`
  `user: user`
  `password: password`
  `stats_interval: 300`
`record:`
  `expire_interval: 10`
`go2rtc:`
  `streams:`
    `driveway:`
     ` - rtsp://user:pass@192.168.1.95:554/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif`
    `driveway_sub:`
     ` - rtsp://user:pass@192.168.1.95:554/cam/realmonitor?channel=1&subtype=2&unicast=true&proto=Onvif`
    `garage_rear:`
     ` - rtsp://user:pass@192.168.1.94:554/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif`
    `garage_rear_sub:`
     ` - rtsp://user:pass@192.168.1.94:554/cam/realmonitor?channel=1&subtype=2&unicast=true&proto=Onvif`
`timestamp_style:`
 ` position: tr`
  `format: "%m/%d/%Y %H:%M:%S"`
  `color:`
   ` red: 255`
    `green: 255`
    `blue: 255`
  `thickness: 2`
  `effect: shadow`
`model:`
 ` path: /home/user/docker-compose/frigate/efficientdet_lite3_512_ptq_edgetpu.tflite`
 ` labelmap_path: /home/user/docker-compose/frigate/coco_labels.txt`
 ` width: 512`
 ` height: 512`

**Reply by skrashevich** [reply]
It is necessary to use paths relative to the container in the configuration, not the host machine. It seems that the path `/home/user/docker-compose/frigate` is related to the host machine, not the container.
For example, if you launch a container with the option `-v /home/user/docker-compose/frigate:/config`, then you need to specify the path in config.yml as `/config/coco_labels.txt` and `/config/efficientdet_lite3_512_ptq_edgetpu.tflite`.

**Reply by superdong69** [reply]
Understood.  How do I go about finding out the paths for my installation? Sorry,  not a Linux guru

**Reply by skrashevich** [reply]
show contents of your `docker-compose.yml` file

**Reply by superdong69** [reply]
`version: "3.9"`
`services:`
  `frigate:`
    `container_name: frigate`
    `privileged: true # this may not be necessary for all setups`
    `restart: unless-stopped`
    `image: ghcr.io/blakeblackshear/frigate:stable`
    `shm_size: "128mb" # update for your cameras based on calculation above`
    `devices:`
      `- /dev/bus/usb:/dev/bus/usb`
     ` - /dev/dri/renderD128 # for intel hwaccel, needs to be updated for your hardware`
    `volumes:`
      `- /etc/localtime:/etc/localtime:ro`
      `- ./frigate.yml:/config/config.yml:ro`
      `- /mnt/data:/media/frigate`
      `- type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear`
      `  target: /tmp/cache`
     `   tmpfs:`
    `      size: 1000000000`
   ` ports:`
  `    - "5000:5000"`
 `     - "1935:1935"   # RTMP feeds`
`      - "8554:8554"   # Birdseye In v 0.12`

**Reply by superdong69** [reply]
Searched the entire drive for "/config/config.yml" and that directory doesn't exist. So I have no idea where I'm supposed to put these two files. I've literally installed it on a heavy metal box, on its own running Ubuntu, using the official documentation. The only thing that isn't the same is my mount point for recordings, everything else is by the book default locations.

**Reply by NickM-27** [reply]
the right side of volumes is inside the container so of course it will never be on your host. The left side is the part you care about, `./frigate.yml` says that the file exists right next to where you are calling docker compose up

**Reply by superdong69** [reply]
Yeah that's where the files are.  / home/user/ docker- compose/ frigate. 

In the model code I even tried ./
Nothing seems to be working

**Reply by NickM-27** [reply]
Your frigate config is trying to define the models with the host path which is impossible. It only knows what's inside the container.

**Reply by NickM-27** [reply]
To get frigate to startup just delete the model config.

**Reply by superdong69** [reply]
No I've got it running now,  I'm trying to get the models skrashevich posted to work

**Reply by NickM-27** [reply]
Right, the container has no idea what `/home/user/docker-compose/frigate` and you are also not passing that directory into the container so currently this won't work. You need to pass the label and model files to the container

**Reply by superdong69** [reply]
Ok,  and I've asked a bunch of times now,  HOW do I do that??

**Reply by NickM-27** [reply]
As an example you just add an item to the volume list 
`- /home/user/docker-compose/frigate/efficientdet_lite3_512_ptq_edgetpu.tflite:/models/efficientdet_lite3_512_ptq_edgetpu.tflite`

and then your frigate config would have the model path as /models/efficientdet_lite3_512_ptq_edgetpu.tflite

**Reply by superdong69** [reply]
In the docker-compose.yml file. 

I GOTCHA!

**Reply by superdong69** [reply]
Same thing required for the coco_labels.txt I surmise?

**Reply by skrashevich** [reply]
yes

**Reply by superdong69** [reply]
It started!

**Reply by superdong69** [reply]
> yes

Any object filter settings for car and person you recommend with these models?

**Reply by superdong69** [reply]
I'm detecting on 1080p feeds. Lowest sub resolution I've got.

**Reply by skrashevich** [reply]
It depends on what you wish to achieve. However, in any case, the configuration of filters and sensitivity is highly dependent on the environment and is individual for each case. Default settings cover the most typical situations.

**Reply by skrashevich** [reply]
> I'm detecting on 1080p feeds. Lowest sub resolution I've got.

The `efficientdet_lite3_512_ptq_edgetpu.tflite` model operates with frames of 512x512 pixels. Prior to processing, all frames are resized to this dimension.

**Reply by superdong69** [reply]
Awesome thanks the both of you for the help! I'll let you know how it gets along

**Reply by superdong69** [reply]
> > I'm detecting on 1080p feeds. Lowest sub resolution I've got.
> 
> The `efficientdet_lite3_512_ptq_edgetpu.tflite` model operates with frames of 512x512 pixels. Prior to processing, all frames are resized to this dimension.

Oh. That's interesting.  So a min_area of 5000 and max_area of 100000 would probably not be ideal. Lol

**Reply by NickM-27** [reply]
min and max area are based on the camera resolution not the model size so assuming your camera resolution didn't change it can be the same as before

**Reply by superdong69** [reply]
Rock on! Thank you again!

**Comment by heebo1974** [comment]
If that accurate model is better, why that is not default in Frigate ?

**Reply by NickM-27** [reply]
Different models work better for different users. They were all trained on the same COCO dataset so they are all not that good relative to purpose built models. The higher size will also lead to much higher inference times
